common/src/main/java/com/wynntils/features/inventory/CustomBankPagesFeature.java

```diff
@@ -0,0 +1,215 @@
+
+    private boolean isBankScreen = false;
+    private boolean quickJumping = false;
+    private int currentPage = 1;
+    private int lastPage = MAX_BANK_PAGES;
+    private int pageDestination = 1;
+    private List<Integer> customJumpDestinations;
+
+    @SubscribeEvent
+    public void onScreenInit(ScreenInitEvent e) {
+        if (!(e.getScreen() instanceof AbstractContainerScreen<?> screen)) return;
+        if (!Models.Container.isBankScreen(screen)) return;
+
+        isBankScreen = true;
+
+        currentPage = Models.Container.getCurrentBankPage(screen);
+
+        customJumpDestinations = List.of(
+                buttonOnePage.get(),
+                buttonTwoPage.get(),
+                buttonThreePage.get(),
+                buttonFourPage.get(),
+                buttonFivePage.get(),
+                buttonSixPage.get());
+
+        if (quickJumping && pageDestination > lastPage) {
+            quickJumping = false;
+            pageDestination = currentPage;
+        } else if (pageDestination != currentPage && quickJumping) {
+            jumpToDestination();
+        } else {
+            quickJumping = false;
+        }
+    }
+
+    @SubscribeEvent
+    public void onContainerClose(ContainerCloseEvent.Post e) {
+        isBankScreen = false;
+        currentPage = 1;
+        pageDestination = 1;
+    }
+
+    @SubscribeEvent
+    public void onSlotClicked(ContainerClickEvent e) {
+        if (!isBankScreen) return;
+
+        int slotIndex = e.getSlotNum();
+
+        if (BUTTON_SLOTS.contains(slotIndex)) {
+            int buttonIndex = BUTTON_SLOTS.indexOf(slotIndex);
+            pageDestination = customJumpDestinations.get(buttonIndex);
+            e.setCanceled(true);
+            jumpToDestination();
+        }
+    }
+
+    @SubscribeEvent
+    public void onRenderSlot(SlotRenderEvent.Pre e) {
+        if (!(e.getScreen() instanceof AbstractContainerScreen<?> screen)) return;
+        if (!isBankScreen) return;
+
+        if (BUTTON_SLOTS.contains(e.getSlot().index)) {
+            ItemStack jumpButton = new ItemStack(Items.DIAMOND_AXE);
+            jumpButton.setDamageValue(92);
+
+            CompoundTag jumpTag = jumpButton.getOrCreateTag();
+            jumpTag.putInt("HideFlags", 6);
+            jumpTag.putBoolean("Unbreakable", true);
+            jumpButton.setTag(jumpTag);
+
+            int buttonIndex = BUTTON_SLOTS.indexOf(e.getSlot().index);
+            int buttonDestination = customJumpDestinations.get(buttonIndex);
+
+            jumpButton.setCount(buttonDestination);
+
+            jumpButton.setHoverName(Component.literal(ChatFormatting.GRAY + "Jump to Page " + buttonDestination));
+
+            e.getSlot().set(jumpButton);
+        }
+
+        if (e.getSlot().index == NEXT_PAGE_SLOT && Models.Container.isLastBankPage(screen)) {
+            lastPage = currentPage;
+        }
+    }
+
+    private void jumpToDestination() {
+        quickJumping = true;
```

Comment:
There is a bit of structure here that seems to be longing to be exposed to the world. :-) Basically, this is a glorified switch case based on the value of `currentPage`. Can you help bring that structure forth?